src/charactersheet/services/character/armor_class_service.js

```diff
@@ -1,166 +1,201 @@
-                score = rawDexBonus;
-            } else {
-                // This case would be for armor of type heavy or no type specified
-                score = 0;
+                return dexMod;
             }
-        } else {
-            // This means the character is wearing no armor,
-            // in which case they get all the dex bonus
-            score = rawDexBonus;
         }
-        return score;
-    };
-
-    self.armorClassModifier = async function() {
-        var uuid = CoreManager.activeCore().uuid();
-        var otherStatsResponse = await OtherStats.ps.read({uuid});
-        const otherStats = otherStatsResponse.object;
-        var armorClassModifier = 0;
+        return dexMod;
+    });
 
-        if (otherStats) {
-            armorClassModifier = otherStats.armorClassModifier() ? parseInt(otherStats.armorClassModifier()) : 0;
+    self.armorClassModifier = ko.pureComputed(() => {
+        if (self.otherStats() && self.otherStats().armorClassModifier()) {
+            const modifier = parseInt(self.otherStats().armorClassModifier());
+            return modifier ? modifier : 0;
         }
+        return 0;
+    });
 
-        return armorClassModifier;
+
+    self.setUpSubscriptions = () => {
+        Notifications.armor.added.add(self.updateArmor);
+        Notifications.armor.changed.add(self.updateArmor);
+        Notifications.armor.deleted.add(self.deleteArmor);
+        Notifications.abilityscore.changed.add(self.updateDexterity);
+        Notifications.otherstats.changed.add(self.updateOtherStats);
+        Notifications.coreManager.changing.add(self.clear);
+        self.armorClass.subscribe(() => Notifications.armorClass.changed.dispatch());
     };
 
-    self.getEquippedShieldBonus = function() {
-        if (self.equippedShield()) {
-            var armorClass = parseInt(self.equippedShield().armorClass());
-            return armorClass ? armorClass : 0;
+    self.updateDexterity = (abilityScore) => {
+        if (abilityScore && abilityScore.name() === Fixtures.abilityScores.constants.dexterity.name) {
+            self.dexterity().importValues(abilityScore.exportValues());
         }
-
-        return 0;
     };
 
-    /* Private Methods */
-
-    self.loadAndSetArmors = async function() {
-        var coreUuid = CoreManager.activeCore().uuid();
-        const response = await Armor.ps.list({coreUuid, equipped: true});
-        let armors = response.objects;
-
-        this.setEquippedArmor(armors);
-        this.setEquippedShield(armors);
+    self.updateOtherStats = (otherStats) => {
+        if (otherStats) {
+            self.otherStats().importValues(otherStats.exportValues());
+        }
     };
 
-    self.setEquippedArmor = function(armors) {
-        const equippedArmor = armors.filter((armor) => {
-            return armor.type() != 'Shield';
-        })[0];
-        self.equippedArmor(equippedArmor);
+    self.updateArmor = (armor) => {
+        if (armor) {
```

Comment:
Don't think the null checks are needed for the incoming objects from notifications.